libs/utils/wa_results_collector.py

```diff
@@ -525,6 +526,10 @@ def _get_metric_df(self, workload, metric, tag, kernel, test):
         """
         Common helper for getting results to plot for a given metric
         """
+        lookup = (workload, metric, tag, kernel, test)
```

Comment:
Can we use the same `memoized` attribute defined by devlib, i.e.
```python
from devlib.utils.misc import memoized
```

This will remove a lot of code and improve consistency with other LISA's code paths, e.g.:
https://github.com/ARM-software/lisa/blob/master/libs/utils/analysis/latency_analysis.py#L53